src/Microsoft.Diagnostics.EventFlow.Inputs.EventSource/EventDataExtensions.cs

```diff
@@ -7,24 +7,37 @@
 using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
-using System.Text;
 using System.Diagnostics.Tracing;
 using System.Diagnostics;
+#if !NETSTANDARD1_6
+using System.Runtime.InteropServices;
+#endif
 
 using Microsoft.Diagnostics.EventFlow.Utilities.Etw;
 
 namespace Microsoft.Diagnostics.EventFlow.Inputs
 {
     internal static class EventDataExtensions
     {
+#if !NETSTANDARD1_6
+        [DllImport("Kernel32.dll", CallingConvention = CallingConvention.Winapi)]
+        private static extern void GetSystemTimePreciseAsFileTime(out long filetime);
+#endif
+
         public static EventData ToEventData(this EventWrittenEventArgs eventSourceEvent, IHealthReporter healthReporter, string context)
         {
             Debug.Assert(healthReporter != null);
 
+#if NETSTANDARD1_6
+            DateTime now = DateTime.UtcNow;
```

Comment:
JFYI: netcoreapp 2.0 clr has precise timestamp implemented with `GetSystemTimePreciseAsFileTime`
But netcoreapp 1.0, 1.1  would still have an imprecise timestamp.

I think it should not be solved (i.e. users are encouraged to use 2.0), but it could make sense to add comment about that

Reply:
Good point, will do, thanks!